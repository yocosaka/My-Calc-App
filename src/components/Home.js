import React from 'react';

const Home = () => {
  const text = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa voluptate quia perferendis, et esse dolor officia sit, blanditiis dignissimos, dolore provident quaerat ullam. Debitis, mollitia! Perspiciatis mollitia odit natus explicabo.';
  return (
    <div className="main-content white">
      <h2 className="main-title">Welcome to our page!</h2>
      <p className="main-text">{text}</p>
      <p className="main-text">{text}</p>
    </div>
  );
};

export default Home;
