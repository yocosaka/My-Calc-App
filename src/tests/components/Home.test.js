import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Home from '../../components/Home';

describe('Home', () => {
  test('renders Home component ', () => {
    render(<Home />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('Welcome to our page!')).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
