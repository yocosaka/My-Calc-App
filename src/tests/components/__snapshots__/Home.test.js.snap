// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match with snapshot 1`] = `
<div
  className="main-content white"
>
  <h2
    className="main-title"
  >
    Welcome to our page!
  </h2>
  <p
    className="main-text"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa voluptate quia perferendis, et esse dolor officia sit, blanditiis dignissimos, dolore provident quaerat ullam. Debitis, mollitia! Perspiciatis mollitia odit natus explicabo.
  </p>
  <p
    className="main-text"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa voluptate quia perferendis, et esse dolor officia sit, blanditiis dignissimos, dolore provident quaerat ullam. Debitis, mollitia! Perspiciatis mollitia odit natus explicabo.
  </p>
</div>
`;
